### PR TITLE
Add `onlyForceKillable` option

### PR DESCRIPTION
**Pull request by yaodingyd**
Add an forceKill option so the process can only be killed by force to simulate that kind of process

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -24,7 +24,8 @@ module.exports = opts => {
 	}
 
 	const title = opts.title ? `process.title = '${opts.title}';` : '';
-	const code = `${title} setInterval(() => {}, 1000 * 1000);console.log('ok');`;
+	const forceKill = opts.forceKill ? 'process.on(\'SIGTERM\', () => {console.log(\'I can only be killed by the force!!\')});' : '';
```
Comment (sindresorhus) [inline index.js]: You need to handle `SIGINT` too.

Comment (sindresorhus) [reply index.js]: Would be more readable if you moved the `process.on(\` string into a template literal variable.

File: readme.md [hunk 2]
```diff
@@ -51,6 +51,13 @@ Default: `false`
 
 Let the process continue to live after the main process exits.
 
+##### forceKill
```
Comment (sindresorhus) [inline readme.md]: I would name it something more explicit: `onlyForceKillable`

File: readme.md [hunk 3]
```diff
@@ -51,6 +51,13 @@ Default: `false`
...
+
+Type: `boolean`<br>
+Default: `false`
+
+Process can only be killed by force.
```
Comment (sindresorhus) [inline readme.md]: Should mention what would kill it `SIGKILL`.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by yaodingyd** [comment]
Also can we get rid of node 4?

**Comment by sindresorhus** [comment]
Yes, but I'll do that after merging this, to keep it focused.

File: index.js (hunk 1 above)
Comment (yaodingyd) [reply index.js]: Do you mean like this:

```javascript
const forceKill = opts.onlyForceKillable ? `
		process.on('SIGTERM', () => {console.log('I can only be killed by the force!!')});
		process.on('SIGINT', () => {console.log('I can only be killed by the force!!')});
		` : '';
```

File: test.js [hunk 4]
```diff
@@ -12,3 +13,15 @@ test('noopProcess()', async t => {
 	const pid = await m({title: 'noop-process-1'});
 	t.true(await processExists(pid));
 });
+
+test('forceKill option', async t => {
```
Comment (sindresorhus) [inline test.js]: You did not update this or the PR title. Please try to better review your changes next time ;)

File: index.js (hunk 1 above)
Comment (sindresorhus) [reply index.js]: Yes

**Pull request merged**